src/storage/engine/mod.rs

```diff
@@ -328,13 +329,17 @@ impl<I: Iterator> Cursor<I> {
             scan_mode: mode,
             min_key: None,
             max_key: None,
+            valid: true,
         }
     }
 
     pub fn seek(&mut self, key: &Key, statistics: &mut CFStatistics) -> Result<bool> {
+        self.valid = true;
+
         assert_ne!(self.scan_mode, ScanMode::Backward);
         if self.max_key.as_ref().map_or(false, |k| k <= key.encoded()) {
             self.iter.validate_key(key)?;
+            self.valid = false;
```

Comment:
Also should set valid to false in L357?

Reply:
Not needed. From two aspects:
- it is a seek out of bound caused by real iterator seek, so we should not set the flag.
- when we reach `!self.iter.seek(key)`, the `iter.valid()` should be false, so the next time `cursor.valid()` will be false, which is correct.